### PR TITLE
Test coverage and HumptyError

### PR DESCRIPTION
**Pull request by AlexanderSchuetz97**

### CODE REVIEW

**Comment by AlexanderSchuetz97** [comment]
must go after #6

File: tests/manual_test_bt.rs
Comment (Grinkers) [inline tests/manual_test_bt.rs:1]: Empty file?

File: src/humpty_error.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+use crate::http::request::HttpVersion;
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+use std::io;
+use std::io::ErrorKind;
+
+pub type HumptyResult<T> = Result<T, HumptyError>;
+
+#[derive(Debug)]
+pub enum RequestHeadParsingError {
```
Comment (Grinkers) [inline src/humpty_error.rs:15]: This should probably have `#[non_exhaustive]`

File: src/humpty_error.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+    //TODO make this not shit
+    Debug::fmt(self, f)
+  }
+}
+
+impl Error for RequestHeadParsingError {}
+
+#[derive(Debug)]
+pub enum HumptyError {
```
Comment (Grinkers) [inline src/humpty_error.rs:41]: This should have `#[non_exhaustive]`

File: src/humpty_error.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+use crate::http::method::Method;
+use crate::http::request::HttpVersion;
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+use std::io;
+use std::io::ErrorKind;
+
+pub type HumptyResult<T> = Result<T, HumptyError>;
+
+#[derive(Debug)]
```
Comment (Grinkers) [inline src/humpty_error.rs:14]: Can we also derive Eq, PartialEq?

File: src/humpty_error.rs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+impl Display for RequestHeadParsingError {
+  fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+    //TODO make this not shit
+    Debug::fmt(self, f)
+  }
+}
+
+impl Error for RequestHeadParsingError {}
+
+#[derive(Debug)]
```
Comment (Grinkers) [inline src/humpty_error.rs:40]: Can we also derive Eq, PartialEq?

**Review by Grinkers — CHANGES_REQUESTED** [review]

File: src/humpty_error.rs (hunk 3 above)
Comment (AlexanderSchuetz97) [reply src/humpty_error.rs]: done

File: src/humpty_error.rs (hunk 2 above)
Comment (AlexanderSchuetz97) [reply src/humpty_error.rs]: done

File: src/humpty_error.rs (hunk 1 above)
Comment (AlexanderSchuetz97) [reply src/humpty_error.rs:16]: done

File: tests/manual_test_bt.rs
Comment (AlexanderSchuetz97) [reply tests/manual_test_bt.rs:1]: by mistake, deleted.

**Review by Grinkers — APPROVED** [review]

**Pull request merged**